includes/class-wc-stripe-customer.php

```diff
@@ -441,11 +405,59 @@ public function set_default_source( $source_id ) {
 		return false;
 	}
 
+	/**
+	 * Get the customer saved payment methods using their Stripe ID.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param string $type Type of Stripe PaymentMethods to return (card, sepa_debit, bacs_debit, etc)
+	 *
+	 * @return array List of saved payment methods for the customer
+	 * @throws WC_Stripe_Exception  An exception if the customer ID is not valid.
+	 */
+	public function get_payment_methods( $type ) {
+		if ( ! $this->get_id() ) {
+			return [];
+		}
+
+		$payment_methods = false;
+		if ( 'card' === $type || 'sepa_debit' === $type ) {
+			$payment_methods = get_transient( "stripe_payment_methods_{$type}_{$this->get_id()}" );
+		}
+
+		if ( false === $payment_methods ) {
+			$response = WC_Stripe_API::request(
+				[
+					'customer' => $this->get_id(),
+					'type'     => $type,
+					'limit'    => 100,
+				],
+				'payment_methods',
+				'GET'
+			);
+
+			if ( ! empty( $response->error ) ) {
+				return [];
+			}
+
+			if ( is_array( $response->data ) ) {
```

Comment:
is it possible for `$response->data` to be anything else than an array after checking `$response->error` is empty?